gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStore.java

```diff
@@ -84,6 +84,7 @@ public class MysqlBaseSpecStore extends InstrumentedSpecStore {
   private static final String GET_ALL_STATEMENT = "SELECT spec_uri, spec FROM %s";
   private static final String GET_ALL_URIS_STATEMENT = "SELECT spec_uri FROM %s";
   private static final String GET_ALL_URIS_WITH_TAG_STATEMENT = "SELECT spec_uri FROM %s WHERE tag = ?";
+  private static final String GET_SPECS_BATCH_STATEMENT = "SELECT spec_uri, spec FROM %s ORDER BY spec_uri ASC LIMIT ? OFFSET ?";
```

Comment:
It's possible that we can have two specs with the same spec_uri I believe. We should keep the modified_time ordering part of the statement too so if there are two specs with the same spec_uri so the ordering of the result from the same query stays the same at all times

Reply:
the `spec_uri` is the primary key [here](https://jarvis.corp.linkedin.com/codesearch/result/?name=MysqlBaseSpecStore.java&path=gobblin-elr%2Fgobblin-runtime%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fgobblin%2Fruntime%2Fspec_store&reponame=linkedin%2Fgobblin-elr#91) in `MysqlBaseSpecStore` and in the `MysqlSpecStore` so ordering on spec_uri should be unique. Let me add a comment in the docstring of the paginate function.

Reply:
1. Was there any reason why spec_json was removed from the query string? 
2. Using this query, it will error in the scenario the OFFSET or LIMIT was set to a negative value as MySQL queries cannot handle those and this [if block](https://github.com/apache/gobblin/pull/3640/files#diff-900cc8e4e863a8057e0e808230a2f9c3c169048059d62a5d94ed9beb94360c12L283-L293) handles such cases. We'll need to add something similar in to handle those cases.

Reply:
Ah that's something we should specify in the function parameter values expected and throw an error if we receive input outside of our expected bounds. We shouldn't receive negative values (count = 0 or offset = 0 should be fine)